### PR TITLE
feat(booking-state): state & transitions

### PR DESCRIPTION
**Pull request by deployAt**
Basic state and transition

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -48,6 +71,45 @@ pub mod pallet {
...
+
+			let time = frame_system::Pallet::<T>::block_number();
+
+			let new_booking = BookingConfig {
+				start: time,
+				end: time.saturating_add(10_u32.into()),
+				status: BookingStatus::Created,
+			};
+
+			Booking::<T>::mutate(|booking| {
```
Comment (korzewski) [inline src/lib.rs:86]: For simple value, you can use 
```
Booking::<T>::set(Some(new_booking.clone()))
```

File: src/lib.rs [hunk 2]
```diff
@@ -48,6 +71,45 @@ pub mod pallet {
...
+				*booking = Some(new_booking.clone());
+			});
+
+			Self::deposit_event(Event::<T>::CreateBooking(new_booking.start, new_booking.end, new_booking.status));
```
Comment (korzewski) [inline src/lib.rs:90]: In this case you may have a similar result by putting a single argument:
```
Self::deposit_event(Event::<T>::CreateBooking(new_booking));
```

File: src/lib.rs [hunk 3]
```diff
@@ -29,10 +47,15 @@ pub mod pallet {
 	#[pallet::getter(fn something)]
 	pub type Something<T> = StorageValue<_, u32>;
 
+	#[pallet::storage]
+	#[pallet::getter(fn booking)]
+	pub type Booking<T: Config> = StorageValue<_, BookingConfig<T::BlockNumber, BookingStatus>>;
```
Comment (korzewski) [inline src/lib.rs:52]: Maybe we can have `BookingConfigOf<T>` for simplification?

**Review by korzewski — APPROVED** [review]
Just small suggestions, all looks good to start with :)

**Comment by deployAt** [comment]
thx

File: src/lib.rs (hunk 2 above)
Comment (deployAt) [reply src/lib.rs:90]: doesn't emit an event this way.

Comment (korzewski) [reply src/lib.rs:90]: hmm maybe I wasn't clear enough. Did you refactor 
Event -> `CreateBooking(BookingConfig)`
as well?

**Pull request merged**